addons/mail/static/src/owl/components/message/message.js

```diff
@@ -408,7 +408,7 @@ class Message extends Component {
      * @param {string} param0.model
      */
     _redirect({ id, model }) {
-        this.trigger('o-redirect', { id, model });
+        this.storeDispatch('redirect', { id, model });
```

Comment:
If you look at why it didn't work, it's because the event `o-redirect` was intercepted by chat window manager and discuss component, both of which basically do a store dispatch. Chatter was broken because no one intercepted the event.

This fix is correct, but you can go further and replace all `.trigger('o-redirect')` in this file by `._redirect()`, and remove the intercepts of `o-redirect`.